content/docs/2.0/scalers/metrics-api.md

```diff
@@ -0,0 +1,95 @@
++++
+title = "Metrics API"
+layout = "scaler"
+availability = "v2.0+"
+maintainer = "Community"
+description = "Scale applications based on a metric value from an API"
+go_file = "metrics_api_scaler"
++++
+
+This specification describes the `metrics-api` trigger that scales based on a metric value from an API.
+
+### Trigger prerequisites
+
+To use an external API as a source of metric you need to have a following endpoint:
+
+```yaml
+/api/v1/metric/{metric}:
+  get:
+    summary: Get Metric
+    description: Get value for a given metric
+    operationId: GetMetric
+    parameters: 
+     - name: metric
+       in: path
+       required: true
+       schema:
+          type: string
+       description: Name of the metric for which information is requested
+    responses:
+      '200':
+        description: Metric inforation was provided
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ReportedMetric'
+      '404':
+        description: Metric is not supported
+      '401':
+        $ref: '#/components/responses/UnauthorizedError'
+      '500':
+        description: Unable to determine metric information
+    tags:
+      - Metrics
+    security:
+      - basic_auth: []
+      - api_key: []
+```
+which returns:
+```yaml
+ReportedMetric:
+  type: object
+  required:
+    - name
+    - value
+  properties:
+    name:
+      type: string
+    value:
+      type: number
+```
+
+
+### Trigger Specification
+
+The metrics API requires following configuration values:
+- `apiURL` string repesenting the base URL of the API (i.e `http://app:1317/api/v1/`).
+- `metricName` the metric name to be used to retive value .
+- `targetValue` the target value for HPA. As long as the current metric doesn't match `TargetValue`, 
+  HPA will increase the number of the pods until it reaches the maximum number of pods allowed to scale to.
+
+
+An example of scaled object using metric API:
+
+```yaml
+apiVersion: keda.k8s.io/v1alpha1
+kind: ScaledObject
+metadata:
+  name: metrics-api-scaledobject
+  namespace: default
+  labels:
+    deploymentName: dummy
+spec:
+  maxReplicaCount: 4
+  scaleTargetRef:
+    deploymentName: dummy
+  triggers:
+    - type: metric-api
+      metadata:
+        targetValue: "1"
+        apiURL: "http://api:3232/api/v1"
```

Comment:
I would just go for `url` here since it's the metrics API scaler
```suggestion
        url: "http://api:3232/api/v1"
```

Reply:
Another question - Should they configure `http://api:3232` or `http://api:3232/api/v1`? Given we've defined the operations and the endpoint might evolve over time I'd argue that it should just be `http://api:3232` and we'll handle the rest, no?

Reply:
What do you think @zroubalik @ahmelsayed ?

Reply:
@tomkerkhove that means that we have to incorporate the API version somewhere else, right?

Reply:
Given they have to comply with our OpenAPI spec it's already covered.

Reply:
https://github.com/kedacore/keda-docs/pull/239/files#diff-2079ef1d5bc7dbe2c2470f43d8749c87R17

Reply:
If that endpoint doesn't exist, then the API is not supported 🤷‍♂️.

If we introduce a `v2

Reply:
I like the current approach, I am not sure if there's a benefit in splitting those fields (the config gets more complex)

Reply:
But if I update my API version in a backward-compatible manner then the endpoint will exist. I think it's better to have full base link as this allows more extensibility. For example, some may have `http://app:1234/experimental/api/v1/` due to ongoing migration process or whatever

Reply:
I'm not saying it has to be separated but just that we care about the base URL.

All metric APIs need to comply with the OpenAPI spec; if we add endpoints there, KEDA will determine what the best one is to call. If other endpoints exist next to it, we'll simply ignore them.

If we configure the full prefix in it, what's the point of having an OpenAPI spec anyway :)

Reply:
>I'm not saying it has to be separated but just that we care about the base URL.

What you mean by "base URL"?

> If we configure the full prefix in it, what's the point of having an OpenAPI spec anyway :)

To be honest, that's the advantage. If you use OpenAPI spec that's good. If you don't... well you can still use the scaler. And in my opinion, that's an advantage. Moreover, we only use `api/v1//metric/{name}` in scaler

Reply:
With base URL I was referring to `http://app:1234`, the rest is handled by the scaler which knows `api/v1/metric/{name}` must be there so it just adds it automatically.

Reply:
For me every user has to use that endpoint, if you want to use ´http://app:1234/experimental/api/v1/` then that's a violation imo

Reply:
Btw. here's @zroubalik comment https://github.com/kedacore/keda-docs/pull/239#discussion_r477143636

If we use only `api/v1/metric/{name}` then how users can change the API version they want to use? And the version can change.

> if you want to use ´http://app:1234/experimental/api/v1/` then that's a violation imo

I would call this violation an "extensibility". Let assume that base url is `http://app:1234`. Should we validate it to do not allow `http://app:1234/staging/user/test/and/more/` as input? 😉